tests/micro/common/test_mlperftiny.py

```diff
@@ -0,0 +1,295 @@
+        import tflite.Model
+
+        tflite_model = tflite.Model.Model.GetRootAsModel(tflite_model_buf, 0)
+
+    interpreter = tf.lite.Interpreter(model_path=str(model_path))
+    interpreter.allocate_tensors()
+    input_details = interpreter.get_input_details()
+    output_details = interpreter.get_output_details()
+
+    model_info = {
+        "input_name": input_details[0]["name"],
+        "input_shape": tuple(input_details[0]["shape"]),
+        "input_dtype": np.dtype(input_details[0]["dtype"]).name,
+        "output_name": output_details[0]["name"],
+        "output_shape": tuple(output_details[0]["shape"]),
+        "output_dtype": np.dtype(output_details[0]["dtype"]).name,
+    }
+
+    if model_name != "ad":
+        model_info["quant_output_scale"] = output_details[0]["quantization_parameters"]["scales"][0]
+        model_info["quant_output_zero_point"] = output_details[0]["quantization_parameters"][
+            "zero_points"
+        ][0]
+
+    relay_mod, params = relay.frontend.from_tflite(
+        tflite_model,
+        shape_dict={model_info["input_name"]: model_info["input_shape"]},
+        dtype_dict={model_info["input_name"]: model_info["input_dtype"]},
+    )
+    return relay_mod, params, model_info
+
+
+def get_test_data(model_name: str) -> list:
+    sample_url = MLPERF_TINY_MODELS[model_name]["sample"]
+    url = urlparse(sample_url)
+    sample_path = download_testdata(sample_url, os.path.basename(url.path), module="data")
+    return [np.load(sample_path)]
+
+
+def predict_ad_labels_aot(session, aot_executor, input_data, runs_per_sample=1):
+    """A special version of tvm/micro/testing/evaluation.py's predict_labels_aot.
+    The runtime returned for each sample is the median of the runtimes for all slices
+    in that sample."""
+
+    assert runs_per_sample > 0
+    assert aot_executor.get_num_inputs() == 1
+    assert aot_executor.get_num_outputs() == 1
+
+    sample_counter = 0
+    for sample in input_data:
+        output_fp32 = np.empty_like(sample)
+        slice_runtimes = []
+
+        for i, time_slice in enumerate(sample):
+            aot_executor.get_input(0).copyfrom(time_slice.reshape((1, 640)))
+            result = aot_executor.module.time_evaluator(
+                "run", session.device, number=runs_per_sample
+            )()
+            slice_runtimes.append(result.mean)
+            output_fp32[i, :] = aot_executor.get_output(0).numpy()
+
+        sample_counter += 1
+        errors = np.mean(np.square(sample - output_fp32), axis=1)
+        yield np.mean(errors), np.median(slice_runtimes)
+
+
+@pytest.mark.parametrize("model_name", ["kws", "vww", "ad", "ic"])
+@pytest.mark.parametrize("project_type", ["host_driven", "mlperftiny"])
+@tvm.testing.requires_micro
+@pytest.mark.skip_boards(
+    ["mps2_an521", "mps3_an547", "stm32f746g_disco", "nucleo_f746zg", "nrf5340dk_nrf5340_cpuapp"]
+)
+def test_mlperftiny_models(platform, board, workspace_dir, serial_number, model_name, project_type):
```

Comment:
add docstring for test (maybe note what it does and doesn't test)

Reply:
done